toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -193,11 +197,37 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 	)
 
 	logBaseName := filepath.Base(srpmFile) + ".log"
+	const checkAttempts = 3
+	totalAttempts := buildAttempts
+	if agent.Config().RunCheck && totalAttempts < checkAttempts {
+		totalAttempts = checkAttempts
+	}
 	err = retry.Run(func() (buildErr error) {
 		builtFiles, logFile, buildErr = agent.BuildPackage(srpmFile, logBaseName, outArch, dependencies)
+		// If the package builds with no errors and RUN_CHECK=y, check logs to see if the %check section passed, and if not, return as the build error.
+		if buildErr == nil && agent.Config().RunCheck {
+			file, logErr := os.Open(logFile)
+			// If we can't open the log file, that's the build error.
+			if logErr != nil {
+				buildErr = logErr
+			} else {
+				scanner := bufio.NewScanner(file)
+				for scanner.Scan() {
+					currLine := scanner.Text()
+					// Anything besides 0 is failed tests
+					if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+						buildErr = errors.New(currLine)
+						if os.Rename(logFile, fmt.Sprintf("%s-fail-%d.log", filepath.Base(srpmFile), time.Now().UnixMilli())) != nil {
+							logger.Log.Debugf("Log file rename failed")
+						}
+						break
+					}
+				}
+			}
+			file.Close()
+		}
```

Comment:
A few clean-up steps:
- Let's always use `defer` to close files and set it up immediately after a successful `os.Open` call. Look for `defer file.Close()` in other places for examples.
- Minimizing the number of indents tends to lead to a code, that's easier to read. Also, we tend to immediately leave the function after encountering an error, thus my suggested change in the `if logErro != nil` logic.
- We can simplify using the scanner a bit.
- Let's add some more debug logging - this is of immense help, when something goes wrong.
- Let's keep and print the re-name error.
- Please extract the log file parsing logic to a separate function. This function is getting a bit bulky.
- Extra credit: add tests for the log parsing function.

**NOTE**: the code change below might not compile, although I did my best. It's meant to visualize my points above.;)
```suggestion
		// If the package builds with no errors and RUN_CHECK=y, check logs to see if the %check section passed, and if not, return as the build error.
		if buildErr != nil {
			return
		}

		if agent.Config().RunCheck {
			file, logErr := os.Open(logFile)
			// If we can't open the log file, that's a build error.
			if logErr != nil {
				logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logfile, logErr)
				buildErr = logErr				
				return
			}
			defer file.Close()

			for scanner := bufio.NewScanner(file); scanner.Scan() {
				currLine := scanner.Text()
				// Anything besides 0 is a failed test
				if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
					failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", filepath.Base(srpmFile), time. Now().UnixMilli())
					buildErr = os.Rename(logfile, failedLogFile)
					if buildErr != nil {
						logger.Log.Errorf("Log file rename failed. Error: %v", buildErr)
						return
					}

					buildErr = fmt.Errorf("package test failed. Test status line: %s", currLine)
					break
				}
			}
```